x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -71,63 +73,86 @@
 
 public class LoggingAuditTrail extends AbstractComponent implements AuditTrail, ClusterStateListener {
 
+    // WARNING: changing any of this requires changing the log4j2.properties file too
+    public static final String REST_ORIGIN_FIELD_VALUE = "rest";
+    public static final String LOCAL_ORIGIN_FIELD_VALUE = "local_node";
+    public static final String TRANSPORT_ORIGIN_FIELD_VALUE = "transport";
+    public static final String IP_FILTER_ORIGIN_FIELD_VALUE = "ip_filter";
+    public static final String ORIGIN_TYPE_FIELD_NAME = "origin.type";
+    public static final String ORIGIN_ADDRESS_FIELD_NAME = "origin.address";
+    public static final String NODE_NAME_FIELD_NAME = "node.name";
+    public static final String HOST_ADDRESS_FIELD_NAME = "host.ip";
+    public static final String HOST_NAME_FIELD_NAME = "host.name";
+    public static final String EVENT_CATEGORY_FIELD_NAME = "event.category";
+    public static final String EVENT_TYPE_FIELD_NAME = "event.type";
+    public static final String EVENT_ACTION_FIELD_NAME = "event.action";
+    public static final String PRINCIPAL_FIELD_NAME = "user.name";
+    public static final String PRINCIPAL_RUN_BY_FIELD_NAME = "user.run_by.name";
+    public static final String PRINCIPAL_RUN_AS_FIELD_NAME = "user.run_as.name";
+    public static final String PRINCIPAL_REALM_FIELD_NAME = "user.realm";
+    public static final String PRINCIPAL_RUN_BY_REALM_FIELD_NAME = "user.run_by.realm";
+    public static final String PRINCIPAL_RUN_AS_REALM_FIELD_NAME = "user.run_as.realm";
+    public static final String PRINCIPAL_ROLES_FIELD_NAME = "user.roles";
+    public static final String REALM_FIELD_NAME = "realm";
+    public static final String URL_PATH_FIELD_NAME = "url.path";
+    public static final String URL_QUERY_FIELD_NAME = "url.query";
+    public static final String REQUEST_BODY_FIELD_NAME = "request.body";
+    public static final String ACTION_FIELD_NAME = "action";
+    public static final String INDICES_FIELD_NAME = "indices";
+    public static final String REQUEST_NAME_FIELD_NAME = "request.name";
+    public static final String TRANSPORT_PROFILE_FIELD_NAME = "transport.profile";
+    public static final String RULE_FIELD_NAME = "rule";
+    public static final String OPAQUE_ID_FIELD_NAME = "opaque_id";
```

Comment:
These are the keys for the `StringMapMessage` and also the field names of the audit event.
If a field name is changed the modification has to be mirrored in the `log4j2.properties` file.
Field names should be [ecs](https://github.com/elastic/ecs) compliant.
@ruflin can you please take a look at them. I know `timestamp` should be `@timestamp` but I wish to change them in one go, so I would appreciate if you can put the stamp of approval over them 😃 
I will pull out a sample of the generated audit trail to make it easy to reason what each field should hold.